### PR TITLE
feat: add login api and authentication

### PR DESCRIPTION
**Pull request by kshitij-k-osmosys**
## Description

- Add authentication in all APIs

- Used `.env` configuration which stores credentials
- Set jwt token to not expire
- Improved readme steps for docker

## Summary by CodeRabbit

- **New Features**
	- Introduced user authentication with JWT token generation and validation.
	- Added authorization requirements for PDF and Word document generation endpoints.
	- Implemented environment variable management for secure application configuration.

- **Documentation**
	- Updated README with detailed setup instructions for environment configuration and Docker usage.

- **Security Enhancements**
	- Added `.env` file to `.gitignore` to prevent sensitive data exposure.
	- Configured Swagger for secure API documentation with JWT support.

- **Bug Fixes**
	- Ensured secure handling of environment variables by stripping quotes and setting them properly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,7 +26,9 @@
 2. [Install Docker Compose](https://docs.docker.com/compose/install/). A separate installation is required for Linux-based OS. If you are using Windows or macOS, installing the Docker Desktop app includes Docker Compose.
 3. Clone the project `document-service`.
 4. (Optional) [Install Docker Extension for VS Code](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-docker).
-5. Set `environment` variables `ASPNETCORE_ENVIRONMENT` and `BUILD_CONFIGURATION` as per requirement in [docker-compose.yaml](./docker-compose.yaml). Ensure correct formatting:
+5. In root directory of project, create a new file `.env`.
+6. Copy data from [example template](.env.example) into `.env`. Then set suitable credentials.
+7. Set `environment` variables `ASPNETCORE_ENVIRONMENT` and `BUILD_CONFIGURATION` as per requirement in [docker-compose.yaml](./docker-compose.yaml). Ensure correct formatting:
```
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply README.md:31]: It is intentional markdown formatting

File: README.md [hunk 2]
```diff
@@ -46,15 +48,16 @@ Setting up the app in a Docker-based environment enables developers of non-Windo
       - BUILD_CONFIGURATION=Release
 ```
 
-6. Ensure Docker is running. Then, in root directory of project, execute the following command to build container for `document-service`:
+8. Ensure Dockeris running.
```
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply README.md]: fixed in latest commit

File: DocumentService.API/Models/LoginRequest.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace DocumentService.API.Models;
+
+public class LoginRequest
```
Comment (sameer-s-b) [inline DocumentService.API/Models/LoginRequest.cs:5]: We don't need a login request for document service.
The idea is to setup a authentication token in the .env file. We will share this token with those who want to use our service.
The token should be of bearer token.

**Review by sameer-s-b — CHANGES_REQUESTED** [review]

File: DocumentService.API/Models/LoginRequest.cs (hunk 3 above)
Comment (kshitij-k-osmosys) [reply DocumentService.API/Models/LoginRequest.cs]: Followed the approach of having a preset JWT_KEY.
All APIs need authentication token for it

File: README.md [hunk 4]
```diff
@@ -26,7 +26,9 @@ Setting up the app in a Docker-based environment enables developers of non-Windo
 2. [Install Docker Compose](https://docs.docker.com/compose/install/). A separate installation is required for Linux-based OS. If you are using Windows or macOS, installing the Docker Desktop app includes Docker Compose.
 3. Clone the project `document-service`.
 4. (Optional) [Install Docker Extension for VS Code](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-docker).
-5. Set `environment` variables `ASPNETCORE_ENVIRONMENT` and `BUILD_CONFIGURATION` as per requirement in [docker-compose.yaml](./docker-compose.yaml). Ensure correct formatting:
+5. In root directory of project, create a new file `.env`.
```
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply README.md]: fixed

File: README.md [hunk 5]
```diff
@@ -46,15 +48,16 @@
       - BUILD_CONFIGURATION=Release
 ```
 
-6. Ensure Docker is running. Then, in root directory of project, execute the following command to build container for `document-service`:
+8. Ensure Docker is running.
+9. Then, in root directory of project, execute the following command to build container for `document-service`:
```
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply README.md]: fixed

File: DocumentService.API/Helpers/AuthenticationHelper.cs [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+        JwtSecurityToken tokenOptions = new JwtSecurityToken(
+            claims: new List<Claim>()
+            {
+                new(ClaimTypes.Email, LoginEmail),
+            },
+            signingCredentials: signinCredentials
+        );
+
+        return new JwtSecurityTokenHandler().WriteToken(tokenOptions);
```
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply DocumentService.API/Helpers/AuthenticationHelper.cs:26]: We can implement this at a later time depending on the requirement

File: README.md [hunk 7]
```diff
@@ -26,7 +26,9 @@ Setting up the app in a Docker-based environment enables developers of non-Windo
 2. [Install Docker Compose](https://docs.docker.com/compose/install/). A separate installation is required for Linux-based OS. If you are using Windows or macOS, installing the Docker Desktop app includes Docker Compose.
 3. Clone the project `document-service`.
 4. (Optional) [Install Docker Extension for VS Code](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-docker).
-5. Set `environment` variables `ASPNETCORE_ENVIRONMENT` and `BUILD_CONFIGURATION` as per requirement in [docker-compose.yaml](./docker-compose.yaml). Ensure correct formatting:
+5. In root directory of project, create a new file `.env`.
+6. Copy data from [example template](.env.example) into `.env`. Then set suitable credentials. Use a valid email. Ensure password has 8-20 characters with 1 uppercase, 1 lowercase, 1 number & 1 special character.
+7. Set `environment` variables `ASPNETCORE_ENVIRONMENT` and `BUILD_CONFIGURATION` as per requirement in [docker-compose.yaml](./docker-compose.yaml). Ensure correct formatting:
 
 #### Development (Default Configuration)
 ```yaml
```
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply README.md]: no need, it is called as this

File: README.md [hunk 8]
```diff
@@ -26,7 +26,9 @@ Setting up the app in a Docker-based environment enables developers of non-Windo
 2. [Install Docker Compose](https://docs.docker.com/compose/install/). A separate installation is required for Linux-based OS. If you are using Windows or macOS, installing the Docker Desktop app includes Docker Compose.
 3. Clone the project `document-service`.
 4. (Optional) [Install Docker Extension for VS Code](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-docker).
-5. Set `environment` variables `ASPNETCORE_ENVIRONMENT` and `BUILD_CONFIGURATION` as per requirement in [docker-compose.yaml](./docker-compose.yaml). Ensure correct formatting:
+5. In root directory of project, create a new file `.env`.
+6. Copy data from [example template](.env.example) into `.env`. Then set suitable credentials.
+7. Set `environment` variables `ASPNETCORE_ENVIRONMENT` and `BUILD_CONFIGURATION` as per requirement in [docker-compose.yaml](./docker-compose.yaml). Ensure correct formatting:
```
[the remark this answers is not in the document]
Comment (kshitij-k-osmosys) [reply README.md]: fixed

**Review by sameer-s-b — APPROVED** [review]

**Pull request merged**